### PR TITLE
Release 2.1.0

### PR DESCRIPTION
**Pull request by ericpre**
Reference: https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/releasing_guide.md

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -10,6 +10,66 @@ https://hyperspy.readthedocs.io/en/latest/changes.html
 
 .. towncrier release notes start
 
+2.1.0 (2024-05-08)
+==================
+
+Enhancements
+------------
+
+- Add an dynamic navigator which updates when the number of navigation dimensions is greater than 3 (`#3199 <https://github.com/hyperspy/hyperspy/issues/3199>`_)
```
Comment (jlaehne) [inline CHANGES.rst]:
```suggestion
- Add a dynamic navigator which updates when the number of navigation dimensions is greater than 3 (`#3199 <https://github.com/hyperspy/hyperspy/issues/3199>`_)
```

File: CHANGES.rst [hunk 2]
```diff
@@ -10,6 +10,66 @@ https://hyperspy.readthedocs.io/en/latest/changes.html
...
+  - Update images in :func:`~.api.plot.plot_images` on data change event.
+  - Remove ROI widgets and disconnect interactive function when closing ROI map figures. (`#3364 <https://github.com/hyperspy/hyperspy/issues/3364>`_)
+- Documentation improvement. (`#3365 <https://github.com/hyperspy/hyperspy/issues/3365>`_)
+
+
+Bug Fixes
+---------
+
+- Fix ROI slicing of non-uniform axis (`#3328 <https://github.com/hyperspy/hyperspy/issues/3328>`_)
+- Adds the ability to save and load :class:`~.axes.BaseDataAxis` objects to a hyperspy file. (`#3342 <https://github.com/hyperspy/hyperspy/issues/3342>`_)
```
Comment (jlaehne) [inline CHANGES.rst]:
```suggestion
- Add the ability to save and load :class:`~.axes.BaseDataAxis` objects to a hyperspy file. (`#3342 <https://github.com/hyperspy/hyperspy/issues/3342>`_)
```

File: CHANGES.rst [hunk 3]
```diff
@@ -10,6 +10,66 @@ https://hyperspy.readthedocs.io/en/latest/changes.html
...
+
+- :func:`~.api.plot.plot_roi_map` doesn't return the sum of all ROI maps (``all_sum``) and the signals sliced by the ROIs (``roi_signals``), these can be obtained separately using the ``rois`` returned by :func:`~.api.plot.plot_roi_map` and :func:`~.api.interactive`. (`#3364 <https://github.com/hyperspy/hyperspy/issues/3364>`_)
+
+
+Maintenance
+-----------
+
+- Ruff update:
+
+  - set the ``RELEASE_next_patch`` branch as target for the ``pre-commit.ci`` update to keep all branches in synchronisation.
```
Comment (jlaehne) [inline CHANGES.rst]:
```suggestion
  - Set the ``RELEASE_next_patch`` branch as target for the ``pre-commit.ci`` update to keep all branches in synchronisation.
```

File: CHANGES.rst [hunk 4]
```diff
@@ -10,6 +10,66 @@ https://hyperspy.readthedocs.io/en/latest/changes.html
...
+  - update ruff to version 0.3.3 and run ruff check/format on source code. (`#3335 <https://github.com/hyperspy/hyperspy/issues/3335>`_)
```
Comment (jlaehne) [inline CHANGES.rst]:
```suggestion
  - Update ruff to version 0.3.3 and run ruff check/format on source code. (`#3335 <https://github.com/hyperspy/hyperspy/issues/3335>`_)
```

File: CHANGES.rst [hunk 5]
```diff
@@ -10,6 +10,66 @@ https://hyperspy.readthedocs.io/en/latest/changes.html
...
 
+2.1.0 (2024-05-08)
+==================
+
+Enhancements
+------------
+
+- Add an dynamic navigator which updates when the number of navigation dimensions is greater than 3 (`#3199 <https://github.com/hyperspy/hyperspy/issues/3199>`_)
+- Add an example to the gallery to show how to extract a line profile from an image using a :class:`~.api.roi.Line2DROI` (`#3227 <https://github.com/hyperspy/hyperspy/issues/3227>`_)
+- Add example showing how to handle RGB images. (`#3346 <https://github.com/hyperspy/hyperspy/issues/3346>`_)
```
Comment (jlaehne) [inline CHANGES.rst]:
```suggestion
- Add an example showing how to handle RGB images. (`#3346 <https://github.com/hyperspy/hyperspy/issues/3346>`_)
```

**Review by jlaehne — APPROVED** [review]

**Comment by ericpre** [comment]
https://pypi.org/project/hyperspy/2.1.0

**Pull request merged**